drivers/iio/light/cm36283.c

```diff
@@ -0,0 +1,852 @@
+	struct i2c_client *client = chip->client;
+	int ret = 0, old_value, *old_thresh;
+
+	if (chan->type != IIO_PROXIMITY)
+		return -EINVAL;
+
+	switch (dir) {
+	case IIO_EV_DIR_FALLING:
+		if (val > chip->ps_close || val < 0)
+			return -EINVAL;
+		old_value = chip->ps_away;
+		chip->ps_away = val;
+		old_thresh = &chip->ps_away;
+		break;
+	case IIO_EV_DIR_RISING:
+		if (val < chip->ps_away || val > chip->max_ps_value)
+			return -EINVAL;
+		old_value = chip->ps_close;
+		chip->ps_close = val;
+		old_thresh = &chip->ps_close;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	mutex_lock(&chip->lock);
+
+	ret = cm36283_set_prox_thresh(chip);
+	if (ret < 0) {
+		dev_err(&client->dev, "Failed to set PS threshold: %d\n", ret);
+		*old_thresh = old_value;
+	}
+
+	mutex_unlock(&chip->lock);
+
+	return ret;
+}
+
+static int cm36283_read_prox_event_config(struct iio_dev *indio_dev,
+					  const struct iio_chan_spec *chan,
+					  enum iio_event_type type,
+					  enum iio_event_direction dir)
+{
+	struct cm36283_data *chip = iio_priv(indio_dev);
+
+	if (chan->type != IIO_PROXIMITY)
+		return -EINVAL;
+
+	switch (dir) {
+	case IIO_EV_DIR_FALLING:
+		return FIELD_GET(CM36283_PS_INT_OUT, chip->ps_conf[CM36283_PS_CONF1]);
+	case IIO_EV_DIR_RISING:
+		return FIELD_GET(CM36283_PS_INT_IN, chip->ps_conf[CM36283_PS_CONF1]);
+	default:
+		return -EINVAL;
+	}
+}
+
+static int cm36283_write_prox_event_config(struct iio_dev *indio_dev,
+					   const struct iio_chan_spec *chan,
+					   enum iio_event_type type,
+					   enum iio_event_direction dir,
+					   bool state)
+{
+	struct cm36283_data *chip = iio_priv(indio_dev);
+	struct i2c_client *client = chip->client;
+	int ret = 0;
+
+	if (chan->type != IIO_PROXIMITY)
+		return -EINVAL;
+
+	int new_ps_conf;
+
+	switch (dir) {
+	case IIO_EV_DIR_FALLING:
+		new_ps_conf = chip->ps_conf[CM36283_PS_CONF1] &
+			      ~CM36283_PS_INT_OUT;
+		new_ps_conf |= FIELD_PREP(CM36283_PS_INT_OUT, state);
+		break;
+	case IIO_EV_DIR_RISING:
+		new_ps_conf = chip->ps_conf[CM36283_PS_CONF1] &
+			      ~CM36283_PS_INT_IN;
+		new_ps_conf |= FIELD_PREP(CM36283_PS_INT_IN, state);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	mutex_lock(&chip->lock);
+
+	ret = i2c_smbus_write_word_data(chip->client, CM36283_REG_PS_CONF1, new_ps_conf);
+	if (ret < 0)
+		dev_err(&client->dev,
+			"Failed to set proximity event interrupt config: %d\n", ret);
+	else
+		chip->ps_conf[CM36283_PS_CONF1] = new_ps_conf;
+
+	mutex_unlock(&chip->lock);
+
+	return ret;
+}
+
+static int cm36283_fallback_read_ps(struct iio_dev *indio_dev)
+{
+	struct cm36283_data *chip = iio_priv(indio_dev);
+	struct i2c_client *client = chip->client;
+	int data = i2c_smbus_read_word_data(client, CM36283_REG_PS_DATA);
+
+	if (data < 0)
+		return data;
+
+	if (data < chip->ps_away)
+		return IIO_EV_DIR_FALLING;
+	else if (data > chip->ps_close)
+		return IIO_EV_DIR_RISING;
+	else
+		return IIO_EV_DIR_NONE;
+}
+
+// Reading the interrupt flag acknowledges the interrupt
```

Comment:
```suggestion
/* Reading the interrupt flag acknowledges the interrupt */
```

I think C style comments are used always, with only exception of the SPDX one due to some weird legacy tooling reasons...


Maybe also worth moving this to the actual read.